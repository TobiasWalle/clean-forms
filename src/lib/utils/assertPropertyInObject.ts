export function assertPropertyInObject(object: any, key: string): void {
  if (!object) {
    throw new TypeError(`Cannot find key "${key}" in ${String(object)}`);
  }
  const objectHasKey = key in object;
  if (!objectHasKey) {
    throw new ReferenceError(`The key "${key}" does not exits on item ${JSON.stringify(object)}.`);
  }
}
