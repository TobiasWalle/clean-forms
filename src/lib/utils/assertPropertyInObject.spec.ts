import { assertPropertyInObject } from './assertPropertyInObject';

describe('assertPropertyInObject', () => {
  it('should throw an error if an object does not have an property', () => {
    const object = { a: 123};
    const property = 'b';

    expect(() => assertPropertyInObject(object, property)).toThrowErrorMatchingSnapshot();
  });

  it('should not throw an error if an object does have an property', () => {
    const object = { a: 123};
    const property = 'a';

    expect(() => assertPropertyInObject(object, property)).not.toThrowError();
  });
});
