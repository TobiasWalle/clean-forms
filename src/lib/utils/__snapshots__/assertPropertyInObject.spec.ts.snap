// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assertPropertyInObject should throw an error if an object does not have an property 1`] = `"The key \\"b\\" does not exits on item {\\"a\\":123}."`;

exports[`assertPropertyInObject should throw an error if object is undefined 1`] = `"Cannot find key \\"b\\" in undefined"`;
